### ISSUE TITLE
[Android] Trying to use this library with wix/react-native-navigation throws error: `undefined is not an object (evaluating 'RNGestureHandlerModule.State')`

### ISSUE THREAD

**Issue by ghoshabhi** [kmagiera/react-native-gesture-handler#128]
Thank you for putting together this awesome library 👍 

I am facing a small issue while integrating it with `react-native-tab-view` and `wix/react-native-navigation`. I raised an issue to [here](https://github.com/react-native-community/react-native-tab-view/issues/474).

I was checking the logs inside Android Studio and get an error saying:

> undefined is not an object (evaluating 'RNGestureHandlerModule.State')

Relevant stack trace:

```
03-05 08:42:30.000 12426-12426/com.tabview_example E/unknown:ReactNative: Unable to display loading message because react activity isn't available

03-05 08:42:31.277 12426-12456/com.tabview_example E/ReactNativeJS: undefined is not an object (evaluating 'RNGestureHandlerModule.State')

03-05 08:42:31.282 12426-12426/com.tabview_example E/unknown:ReactNative: Unable to launch redbox because react activity is not available, here is the error that redbox would've displayed: undefined is not an object (evaluating 'RNGestureHandlerModule.State')

03-05 08:42:32.459 2120-2254/system_process E/TaskPersister: File error accessing recents 
directory (directory doesn't exist?).
```

Here's the example repo with the change: https://github.com/ghoshabhi/rn-tabview-problem/commit/9d5b1eed6c8ab2adf716094e76a868fa44b016b0

When I run the app - I see a blank white screen with no error (you can see that in the first line of the error trace I have shared)

I am wrapping my navigation screens with the `gestureHandlerRootHOC` component too.

Am I missing something here ?

`react-native: v0.54.0`
`react-native-gesture-handler: v1.0.0-alpha.41`
`react-native-tab-view: v0.0.74`
`react-native-navigation: v1.1.397`

**Comment by AndrewJack** [comment]
I've added some comments on the commit you linked.

You haven't added the package to the MainApplication.java 
https://github.com/ghoshabhi/rn-tabview-problem/blob/master/android/app/src/main/java/com/tabview_example/MainApplication.java#L22

That'll probably fix your issue.

**Comment by ghoshabhi** [comment]
@AndrewJack : Thanks for looking into it. I thought `react-native link` would've done it! I will try out manual linking anyways.

**Comment by qasimalbaqali** [comment]
@ghoshabhi I'm getting the same error, have you solved it?

**Comment by ghoshabhi** [comment]
@qasimalbaqali : No I did not get time to try it back out again. Here is a sample repository: https://github.com/ghoshabhi/rn-tabview-problem for you to play around.

If you're using `react-native-tab-view` I got around it using `TabViewPagerPan` from the package itself:

```
 renderPager = (props: any) => <TabViewPagerPan {...props} />;
```

**Comment by Apak00** [comment]
That didn't work for me @AndrewJack am I missing something?
`
import android.app.Application;

import com.facebook.react.ReactApplication;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.soloader.SoLoader;
import com.swmansion.gesturehandler.react.RNGestureHandlerPackage;
import com.oblador.vectoricons.VectorIconsPackage;

import java.util.Arrays;
import java.util.List;

public class MainApplication extends Application implements ReactApplication {

    private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
        @Override
        public boolean getUseDeveloperSupport() {
            return BuildConfig.DEBUG;
        }

        @Override
        protected List<ReactPackage> getPackages() {
            return Arrays.<ReactPackage>asList(
            new MainReactPackage(),
            new VectorIconsPackage(),
            new RNGestureHandlerPackage()
            );
        }

        @Override
        protected String getJSMainModuleName() {
            return "index";
        }
    };

    @Override
    public ReactNativeHost getReactNativeHost() {
        return mReactNativeHost;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        SoLoader.init(this, /* native exopackage */ false);
    }
}`

**Comment by AndrewJack** [comment]
@Apak00 you also need to add some code to the MainActivity - https://github.com/kmagiera/react-native-gesture-handler#installation

```java
// Don't forget imports
import com.facebook.react.ReactActivityDelegate;
import com.facebook.react.ReactRootView;
import com.swmansion.gesturehandler.react.RNGestureHandlerEnabledRootView;

public class MainActivity extends ReactActivity {

  // Add the following method to your main activity class
  @Override
  protected ReactActivityDelegate createReactActivityDelegate() {
    return new ReactActivityDelegate(this, getMainComponentName()) {
      @Override
      protected ReactRootView createRootView() {
        return new RNGestureHandlerEnabledRootView(MainActivity.this);
      }
    };
  }
}
```

**Comment by osdnk** [comment]
This issue seems to be inactive
@ghoshabhi, If you still face any problem, let us know

**Comment by Albert0405** [comment]
Im having this problem. After i install 'react-native-gesture-handler' and link it 

`undefined is not an object (evaluating 'RNGestureHandlerModule.State')` 

I also tried to change what's on my Android MainActivity and MainApplication and im still facing the same problem. The problem gone when i do not import the plugin inside my code

**Comment by 91ranjan** [comment]
Same here with wix/react-native-navigation the MainActiveity needs to be extended with SplashActivity. When i do 
`    @Override
    protected ReactActivityDelegate createReactActivityDelegate() {
        return new ReactActivityDelegate(this, getMainComponentName()) {
            @Override
            protected ReactRootView createRootView() {
                return new RNGestureHandlerEnabledRootView(MainActivity.this);
            }
        };
    }
`
It says @overrideing a function not defined in the super class.

There was another issue with the wix which got resolved recently that would not let the red screen load. I applied that fix too. Just for reference to others.

https://github.com/wix/react-native-navigation/pull/2801#issuecomment-394278656

![whatsapp image 2018-06-07 at 8 01 00 pm](https://user-images.githubusercontent.com/5716845/41107772-a3fa6a08-6a90-11e8-91d9-ac69e174a593.jpeg)


Please let me with any patch or solution.
Also not sure how to open this issue.
@osdnk @AndrewJack

**Comment by AndrewJack** [comment]
Looks like [SplashActivity](https://github.com/wix/react-native-navigation/blob/master/android/app/src/main/java/com/reactnativenavigation/controllers/SplashActivity.java) doesn't extend ReactActivity. 

You should add this method to the activity that extends ReactActivity

**Comment by 91ranjan** [comment]
Still waiting for their answer but for the time being is there any workaround?

Also is the above error only due not able to add this code to Main Activity because of the SplashActivity issue?

**Comment by AndrewJack** [comment]
These issue looks similar #35 & https://github.com/wix/react-native-navigation/issues/1950

**Comment by 91ranjan** [comment]
Yeah ... But there is no solution as I can find there. The file referenced there "https://github.com/kmagiera/react-native-gesture-handler/blob/master/NATIVE_NAVIGATORS.md"
is not more valid.

**Comment by latircoleetti** [comment]
@91ranjan @AndrewJack was this issue ever resolved? Having the same problem now after adding `gestureHandlerRootHOC` to relevant screens.

**Comment by ifrostyice** [comment]
![whatsapp image 2018-11-21 at 22 39 14](https://user-images.githubusercontent.com/41566290/48852495-15437600-ede0-11e8-8c1e-158e4745a0c1.jpeg)
 help

**Comment by OmgDef** [comment]
Same here with the react native navigation v2

**Comment by HamzaIqbal111** [comment]
For React Native, follow these steps

1. remove node_modules and package-lock.json 

2. npm install

3. npm install --save react-navigation

4. npm install --save react-native-gesture-handler

5. react-native link

**Comment by osdnk** [comment]
I'm extremely sure that there are some problems with linking. 
Please follow steps given by @HamzaIqbal111.
If there is still any case, please try to link in "manually" by adding deps via `MainApplication.java`.

**Comment by betasve** [comment]
Exactly the same problem for me. Tried everything described here and on another 100 places on SO. Still the same error. This is getting frustrating.

**Comment by JohnnyDevX** [comment]
Same problem on a very simple Wix Navigation app with the lib (and react-native-tab-view)

**Comment by JohnnyDevX** [comment]
Ok fixed it!
1. In **MainApplication.java**:

    ...
    import com.swmansion.gesturehandler.react.RNGestureHandlerPackage;
    ...
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
        // eg. new VectorIconsPackage()
        new RNGestureHandlerPackage() // <-- Add this!
      );
    }
    ...

2. And follow these steps: https://kmagiera.github.io/react-native-gesture-handler/docs/getting-started.html#with-wix-react-native-navigation-https-githubcom-wix-react-native-navigation

I think we can close the issue.

Thank you for the library Magiera!
